source/tutorial/autocomplete-component.md

```diff
@@ -303,57 +303,63 @@ similar to [how we tested our rental listing component earlier](../simple-compon
+import { render } from '@ember/test-helpers';
 import hbs from 'htmlbars-inline-precompile';
 
-moduleForComponent('list-filter', 'Integration | Component | filter listing', {
-  integration: true
-});
+module('Integration | Component | rental-listing', function(hooks) {
+  setupRenderingTest(hooks);
 
-test('should initially load all listings', function (assert) {
-});
+  test('should initially load all listings', function (assert) {
+  });
 
-test('it renders', function(assert) {
-  // Set any properties with this.set('myProperty', 'value');
-  // Handle any actions with this.on('myAction', function(val) { ... });
+  test('it renders', async function(assert) {
+    // Set any properties with this.set('myProperty', 'value');
+    // Handle any actions with this.set('myAction', function(val) { ... });
 
-  this.render(hbs`{{list-filter}}`);
+    await render(hbs`{{rental-listing}}`);
 
-  assert.equal(this.$().text().trim(), '');
+    assert.equal(this.element.textContent.trim(), '');
 
-  // Template block usage:
-  this.render(hbs`
-    {{#list-filter}}
-      template block text
-    {{/list-filter}}
-  `);
+    // Template block usage:
+    await render(hbs`
+      {{#rental-listing}}
+        template block text
+      {{/rental-listing}}
+    `);
+
+    assert.equal(this.element.textContent.trim(), 'template block text');
+  });
 
-  assert.equal(this.$().text().trim(), 'template block text');
 });
 ```
 
 Our list-filter component takes a function as an argument, used to find the list of matching rentals based on the filter string provided by the user.
 We provide an action function by setting it to the local scope of our test by calling `this.on`.
 
-```tests/integration/components/list-filter-test.js{+3,+5,+6,+13,+14,+15,+16,+17}
-import { moduleForComponent, test } from 'ember-qunit';
+```tests/integration/components/list-filter-test.js{+5,+7,+8,+14,+15,+16,+17,+18}
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
 import hbs from 'htmlbars-inline-precompile';
-import RSVP from 'rsvp';
+import { resolve } from 'rsvp';
 
 const ITEMS = [{city: 'San Francisco'}, {city: 'Portland'}, {city: 'Seattle'}];
 const FILTERED_ITEMS = [{city: 'San Francisco'}];
 
-moduleForComponent('list-filter', 'Integration | Component | filter listing', {
-  integration: true
-});
+module('Integration | Component | rental-listing', function(hooks) {
+  setupRenderingTest(hooks);
 
-test('should initially load all listings', function (assert) {
-  // we want our actions to return promises,
-  //since they are potentially fetching data asynchronously
-  this.on('filterByCity', () => {
-    return RSVP.resolve({ results: ITEMS });
+  test('should initially load all listings', async function (assert) {
+    // we want our actions to return promises,
+    //since they are potentially fetching data asynchronously
+    this.on('filterByCity', () => {
```

Comment:
this could just be `this.on('filterByCity', resolve({results: ITEMS});`

Reply:
I don't think this will work either since `this.on()` is no longer a thing. 

I think the following should work:
```js
this.set('filterByCity', () => resolve(...));
```

```hbs
{{some-component onFoo=(action filterByCity)}}
```

Reply:
sorry the codemod in super-rentals repo did this, which works:
```javascript

  test('should initially load all listings', async function(assert) {
    assert.expect(2);
    this.actions.filterByCity = () => {
      return resolve({ results: ITEMS });
    };

    await render(hbs`
      {{#list-filter filter=(action 'filterByCity') as |rentals|}}
        <ul>
        {{#each rentals as |item|}}
          <li class="city">
            {{item.city}}
          </li>
        {{/each}}
        </ul>
      {{/list-filter}}
    `);

    return settled().then(() => {
      assert.equal(this.element.querySelectorAll('.city').length, 3);
      assert.equal(this.element.querySelector('.city').textContent.trim(), 'San Francisco');
    });
  });
```

Reply:
will update here...

Reply:
to my knowledge the codemod did the `this.actions.<actionName>` because it was safer than potentially blowing away people's properties, not because it is encouraged to write `this.actions`  @rwjblue  to confirm

given that glimmer actions are just functions and will be passed in like `(action filterByCity)` and we will eventually get there, I prefer @Turbo87 's recommendation.

In my tests I use the below to great sucess
```js
this.filterByCity = ( ) => {
 // stuff
}
await render(hbs`{{my-component click=(action filtlerByCity)}`);
```

but I'm weird because I do that also in my components too instead of using the `actions`  hash most of the time

Reply:
ok, will update 👍

Reply:
Yes, confirm RE: the codemod results. Whenever possible, we should use and suggest closure actions.